pkg/controller/workload/job/job_controller.go

```diff
@@ -178,11 +178,8 @@ func SetupIndexes(ctx context.Context, indexer client.FieldIndexer) error {
 		// grab the Workload object, extract the owner...
 		wl := o.(*kueue.Workload)
 		owner := metav1.GetControllerOf(wl)
-		if owner == nil {
-			return nil
-		}
 		// ...make sure it's a Job...
-		if owner.APIVersion != "batch/v1" || owner.Kind != "Job" {
+		if owner == nil || owner.APIVersion != "batch/v1" || owner.Kind != "Job" {
```

Comment:
```suggestion
		if owner == nil || owner.APIVersion != batchv1.SchemeGroupVersion.String() || owner.Kind != "Job" {
```

Can we avoid hardcoding the apiVersion?

Reply:
Absolutely.